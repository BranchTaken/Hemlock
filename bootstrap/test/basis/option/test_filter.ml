open! Basis.Rudiments
open! Basis
open Option
open Format

let test () =
  printf "@[<h>";
  List.iter [false; true] ~f:(fun b ->
    List.iter [Some 42L; None] ~f:(fun o ->
      printf "filter %a ~f:(fun _ -> %b) -> %a\n"
        (pp Uns.pp) o
        b
        (pp Uns.pp) (filter o ~f:(fun _ -> b))
    )
  );
  printf "@]"

let _ = test ()
