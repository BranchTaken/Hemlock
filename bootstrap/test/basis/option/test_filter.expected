filter Some (42) ~f:(fun _ -> false) -> None
filter None ~f:(fun _ -> false) -> None
filter Some (42) ~f:(fun _ -> true) -> Some (42)
filter None ~f:(fun _ -> true) -> None
